packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -719,4 +759,36 @@ contract Bridge is EssentialContract, IBridge {
             return block.timestamp >= _receivedAt.max(lastUnpausedAt) + _invocationDelay;
         }
     }
+
+    /// @notice Calculates the actual processing fee.
+    /// @param _isNewlyProven True if the message is proven in the same transaction.
+    /// @param _gasStartValue The gasleft() value before processing.
+    /// @param _msgGasLimit The message's gas limit.
+    /// @param _msgFee The message's fee.
+    /// @return The processing fee.
+    function _calcFee(
+        bool _isNewlyProven,
+        uint256 _gasStartValue,
+        uint256 _msgGasLimit,
+        uint256 _msgFee
+    )
+        private
+        view
+        returns (uint256)
+    {
+        uint256 gasUsed = _gasStartValue - gasleft();
+        gasUsed += _isNewlyProven // use different overhead value
+            ? ONE_STEP_PROCESSING_GAS_OVERHEAD
+            : TWO_STEP_PROCESSING_GAS_OVERHEAD;
+        uint256 actualCost = block.basefee * gasUsed; // tip is not considered
```

Comment:
Are we sure it is the desired calc (without tip) ? 

The realyer gets only this:
     `msg.sender.sendEtherAndVerify(fee, _SEND_ETHER_GAS_LIM)`  (which is on line 95).

But the fee is calculated with no tip 🤔 Maybe i need to wrap my head around it a little bit more.

Reply:
The fee is added below and it's that fee that is the `fee` value you posted.

Reply:
If we use gas price not base fee, then the relayer can use a really high tip to take all the fees -- imagine the relayer is a validator on L1.

Reply:
Oh that tip, seems like I misunderstood Dani's question. Yeah the relayer will have to make up for the tip he paid to the L1 validator in his L1 tx using the 50% he gets from the max fee set in the message.